mdakits/domhmm/metadata.yaml

```
@@ -0,0 +1,121 @@
+  - biomemphys
+
+## str: a free form description of the mdakit
+description:
+    DomHMM provides an automated workflow to identify liquid-ordered (Lo) domains from Molecular Dynamics simulations of bio-membranes.
+
+## List(str): a list of keywords which describe the mdakit
+keywords:
+  - membranes
+  - molecular dynamics
+  - nanodomains
+  - microdomains
+
+## str: the license the mdakit falls under
+## See https://spdx.org/licenses/ for valid license specifiers
+license: GPL-2.0-or-later
+
+## str: the link to the project's code
+## Please note that this is not limited to GitHub! Can be Gitlab, etc..
+project_home: https://github.com/BioMemPhys-FAU/domhmm
+
+## str: the link to the project's documentation
+documentation_home: https://domhmm.readthedocs.io
+
+## str: the type of documentation available [UserGuide, API, README]
+documentation_type: UserGuide + API
+
+## List(str): a list of commands to use when installing the mdakit from its
+## source code.
+src_install:
+  - pip install git+https://github.com/BioMemPhys-FAU/domhmm@main
+
+## str: the package name used to import the mdakit
+import_name: domhmm
+
+## str: a specification for the range of Python versions supported by this MDAKit
+python_requires: ">=3.9"
+
+## str: a specification for the range of MDAnalysis versions supported by this MDAKit
+mdanalysis_requires: ">=2.0.0"
+
+## List(str): a list of commands to use when attempting to run the MDAKit's tests
+## If you package your tests inside your package then you can typically use the 
+##     pytest --pyargs MYPACKAGE
+## command as shown below. 
+## Otherwise you need to include commands to make the tests available. 
+## For example, if the tests are in the repository at the top level under `./tests`:
+## First use `git clone latest` to either clone the top commit for "development code" checks or check out
+## the latest tag for "latest release" checks. Then then run pytest:
+##    - git clone latest
+##    - pytest -v ./tests
+## Feel free to ask for advice on your pull request!
+run_tests:
+  - pytest --pyargs domhmm
+
+## List(str): a list of commands to use to install the necessary dependencies required
+## to run the MDAKit's tests.
+## The default below _might_ be sufficient or you might not even need MDAnalysisTests:
+## make sure that it is appropriate for how you run tests.
+test_dependencies:
+  - mamba install pytest
+
+## str: the organisation name the MDAKit falls under
+project_org: BioMemPhys-FAU
+
+#------------------------------------------------------------
+# Optional entries
+#------------------------------------------------------------   
+## List(str): a list of commands to use when installing the latest
+## release of the code. Note: only one installation method can currently
+## be defined. We suggest using mamba where possible (e.g.
+##   mamba -c conda-forge install MYPROJECT
+## for a conda package installation).
+## Here we use a simple PyPi installation:
+#install:
+#  - pip install MYPROJECT
```

Your package seems to be in good state already. Would you consider adding it to PyPi? It's a straightforward process and it would make it a lot easier for users to install.

Otherwise we can say here
```yaml
install:
 - git clone latest
 - pip install .
```
(I _think_ this should work, asking @IAlibay for confirmation/suggestions.)

---

Hi @orbeckst ,
I will tag @yusuferentunc . He is handling the installation issues and will look into adding DomHMM to PyPi.

---

Do you want to try out
```suggestion
install:
 - git clone latest
 - pip install .
```
and see if this runs? This would be a first step. You can immediately replace it with
```
install:
 - pip install domhmm
```
once @yusuferentunc has created the PyPi package.

---

I tried the following sequence of commands:

1. git clone https://github.com/BioMemPhys-FAU/domhmm.git
2. cd domhmm
3. pip install .

And it installed domhmm successfully. Is this the workflow you had in mind?

---

Sort of, you also want to get the latest release tag, which is 1.0.0 in your case:
```
git clone https://github.com/BioMemPhys-FAU/domhmm.git
cd domhmm
git fetch --tags
git checkout 1.0.0
pip install .
```
but for the meta.yml's install target you don't need the `cd domhmm` or know the latest tage because that's part of the magic of `git clone latest` — which is not really a git command but a special step that we use and parse in a specific way  https://github.com/MDAnalysis/MDAKits/blob/8a04d6ada4eb65d2b7d0ac3527746e9d5444cbbc/utils/mdakit.py#L540-L546 

So for docs you would write something like the above but for the meta.yml you'd use the commands that I wrote in https://github.com/MDAnalysis/MDAKits/pull/164#discussion_r1752675531 .

---

I tried it for python 3.9, 3.10, and 3.12, and I could install domhmm successfully. I only got an error message for python 3.9 regarding pip on my M1 MacBook: "ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. "However, if I switch to an intel-based Mac, everything works fine.

---

Python 3.9 will not even be supported for MDAnalysis 2.8.0 so I think having some issues with the 3.9 and Apple silicon is acceptable.

---

Hm, sorry, my suggestion did not work — https://github.com/MDAnalysis/MDAKits/actions/runs/10811290602/job/30011929019?pr=164#step:7:19

@IAlibay does `git clone latest` not work with `install:` ?

---

Given that my original idea did not work, you could have it explicitly install the 1.0.0 with
```yaml
install:
  - mamba install -c conda-forge numpy scipy scitkit-learn hmmlearn networkx
  - pip install https://github.com/BioMemPhys-FAU/domhmm/archive/refs/tags/1.0.0.tar.gz
```
(MDAnalysis will be installed automatically.)

Once you have the pypi package up, you can update this entry in the registry.

By the way, I looked at your dependencies in https://github.com/BioMemPhys-FAU/domhmm/blob/de318b9a3c0cd07eb083d5b6d2c944e7a3f6201f/pyproject.toml#L22-L28 and noticed that you're pinning to fixed versions. I would avoid that unless you have a very good scientific reason because pinned packages can lead to mutually exclusive dependencies and make it very difficult for users to create an environment with their own choice of packages. Instead I would use ranges such as `>=0.3.2` or `>=1.0,<2.0` (you need to look up the exact syntax but you get the idea). 

Alternatively, if you don't want to test the 1.0.0 release here, you can also comment out the `install:` part. It is [an optional entry](https://mdakits.mdanalysis.org/reviewersguide.html#optional-entries) for a MDAKit and so I would approve the registration without it. Users would see a gray banner for "Latest Release" in the registry. You could update your registration once you have the PyPi package available.

---

Duplicating what I mentioned on discord: `install` is a specific thing - it indicates "how do I install the packaged version of this code". If the code is only releasing through github, i.e. no package is being distributed, then `src_install` is all you can advertise in the registry, because all that mdakit is offering is source installs.

Please do not hardcode a tag in `install`, that breaks the idea of how the registry should work.

---

The hard-coded tag was meant to be an interim until the pypi package is up.

It sounds to me that the easiest way forward is to **just comment out `install` for now** and update the metadata once you have a pypi package.

(For more discussion on `install` see discord.)

---

Thanks @orbeckst and @IAlibay for your input and your suggestions! I commented out `install`, @yusuferentunc is currently looking into the pypi issue. Unfortunately, he will be not in the office until next week, so maybe it can take a few more days. I opened further an [issue](https://github.com/BioMemPhys-FAU/domhmm/issues/53#issue-2523298903) for the versions of our dependencies and I will take a look as soon as possible.